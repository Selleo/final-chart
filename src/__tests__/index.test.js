import { pipe } from '../index';

const base = (state, props) =>
  state
    .set('title.text', null)
    .set('chart.type', props.chart.type)
    .set('plotOptions.series.stacked', props.chart.stacked);

const series = state =>
  state
    .set('series', []);

const initState = {};

const props = {
  chart: {
    type: 'pie',
    stacked: true,
  }
};

describe('pipe', () => {
  test('base and series', () => {
    expect(pipe(base, series)(initState, props)).toMatchSnapshot();
  });
});
