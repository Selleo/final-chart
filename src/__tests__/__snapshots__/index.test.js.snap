// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipe base and series 1`] = `
Object {
  "chart": Object {
    "type": "pie",
  },
  "plotOptions": Object {
    "series": Object {
      "stacked": true,
    },
  },
  "series": Array [],
  "title": Object {
    "text": null,
  },
}
`;
